Framework/Algorithms/inc/MantidAlgorithms/DiscusMultipleScatteringCorrection.h

```diff
@@ -51,50 +51,68 @@ class MANTID_ALGORITHMS_DLL DiscusMultipleScatteringCorrection : public API::Alg
+  double interpolateGaussian(const API::ISpectrum &histToInterpolate, double x);
+  double Interpolate2D(API::MatrixWorkspace_sptr SOfQ, double w, double q);
   void updateTrackDirection(Geometry::Track &track, const double cosT, const double phi);
-  void integrateCumulative(const Mantid::HistogramData::Histogram &h, double xmax, std::vector<double> &resultX,
-                           std::vector<double> &resultY);
+  void integrateCumulative(const Mantid::HistogramData::Histogram &h, double xmin, double xmax,
+                           std::vector<double> &resultX, std::vector<double> &resultY);
 
 private:
   void init() override;
   void exec() override;
   std::map<std::string, std::string> validateInputs() override;
   API::MatrixWorkspace_sptr createOutputWorkspace(const API::MatrixWorkspace &inputWS) const;
   std::tuple<double, double> new_vector(const Kernel::Material &material, double kinc, bool specialSingleScatterCalc);
-  double simulatePaths(const int nEvents, const int nScatters, Kernel::PseudoRandomNumberGenerator &rng,
-                       const HistogramData::Histogram &invPOfQ, const double kinc, const Kernel::V3D &detPos,
-                       bool specialSingleScatterCalc);
-  std::tuple<bool, double, double> scatter(const int nScatters, Kernel::PseudoRandomNumberGenerator &rng,
-                                           const HistogramData::Histogram &invPOfQ, const double kinc,
-                                           const Kernel::V3D &detPos, bool specialSingleScatterCalc);
+  std::vector<double> simulatePaths(const int nEvents, const int nScatters, Kernel::PseudoRandomNumberGenerator &rng,
+                                    API::MatrixWorkspace_sptr &invPOfQ, const double kinc,
+                                    const std::vector<double> &wValues, const Kernel::V3D &detPos,
+                                    bool specialSingleScatterCalc);
+  std::tuple<bool, std::vector<double>, double> scatter(const int nScatters, Kernel::PseudoRandomNumberGenerator &rng,
+                                                        const API::MatrixWorkspace_sptr &invPOfQ, const double kinc,
+                                                        const std::vector<double> &wValues, const Kernel::V3D &detPos,
+                                                        bool specialSingleScatterCalc);
   Geometry::Track start_point(Kernel::PseudoRandomNumberGenerator &rng);
   Geometry::Track generateInitialTrack(Kernel::PseudoRandomNumberGenerator &rng);
   void inc_xyz(Geometry::Track &track, double vl);
   void updateWeightAndPosition(Geometry::Track &track, double &weight, const double vmfp, const double sigma_total,
                                Kernel::PseudoRandomNumberGenerator &rng);
-  void q_dir(Geometry::Track &track, const HistogramData::Histogram &invPOfQ, const double kinc,
+  void q_dir(Geometry::Track &track, const API::MatrixWorkspace_sptr &invPOfQ, double &k,
```

Comment:
The variable `kinc` in this function has been renamed to just `k` (presumably to reflect that ki or kf could be fixed?) - should this be done in other functions as well?

Reply:
I've had a look through. Some functions such as simulatePaths and scatter do genuinely take the incident k value of a neutron path before it reaches the sample so these two are fine. There are other functions though (eg new_vector, sampleKW) that get called multiple times for a single neutron path (eg at each scatter) and there's a bit of ambiguity in my variable\parameter naming. On some occasions I've also used kinc to refer to the k value before the scatter but if it's the 2nd\3rd scatter this won't be the same as kinc. Perhaps I should call this kbefore to distinguish it from kinc.

Reply:
No I think `kinc` is good when talking about the incident k-vector in a scattering event regardless of the order of the scattering, just would be good to distinguish experimental kfixed from ki/kf of a scattering event - but variable names can only do so  much!